src/Microsoft.AspNetCore.Razor.Design/build/netstandard2.0/Microsoft.AspNetCore.Razor.Design.targets

```diff
@@ -42,45 +42,71 @@
+    <RazorCompileOnBuild>false</RazorCompileOnBuild>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(RazorCompileOnPublish)'==''">
+    <!-- Always compile on publish by default if we're compiling on build -->
+    <RazorCompileOnPublish Condition="'$(RazorCompileOnBuild)'=='true'"/>
+
+    <!-- Compatibility with the old MVC Precompilation setting -->
+    <RazorCompileOnPublish Condition="'$(RazorCompileOnPublish)'==''">$(MvcRazorCompileOnPublish)</RazorCompileOnPublish>
+
+    <!-- Default to on if MvcRazorCompileOnPublish isn't set for some reason -->
     <RazorCompileOnPublish Condition="'$(RazorCompileOnPublish)'==''">true</RazorCompileOnPublish>
+  </PropertyGroup>
 
+
+  <!--
+    Properties that configure Razor SDK, but need to be defined in targets due to evaluation order.
+  -->
+  <PropertyGroup>
     <!-- Output directory used for generated files -->
-    <RazorGenerateOutputPath Condition="'$(RazorGenerateOutputPath)'==''">$(IntermediateOutputPath)Razor\</RazorGenerateOutputPath>
+    <RazorGenerateIntermediateOutputPath Condition="'$(RazorGenerateIntermediateOutputPath)'==''">$(IntermediateOutputPath)Razor\</RazorGenerateIntermediateOutputPath>
     
     <!-- File name (without extension) of the assembly produced by Razor -->
     <RazorTargetName Condition="'$(RazorTargetName)'==''">$(TargetName).PrecompiledViews</RazorTargetName>
 
-    <!-- 
-      Set to true to copy RazorGenerate items to the publish directory (.cshtml) files.
+    <!--
+      The compatibility zone - these properties were provided by the MVC Precompilation tool and they
+      map to supported settings in Razor SDK.
 
-      Typically Razor files are not needed for a published application if they participate in compilation at build-time
-      or publish-time. By default, the Razor SDK will suppress the copying of RazorGenerate items to the publish directory.
+      We want to set the defaults for these in the .props file, but we need to process the old settings here
+      in case they were set in the project file. The consequence of this is that the old settings will override
+      the new ones if they are set to conflicting values.
     -->
-    <CopyRazorGenerateFilesToPublishDirectory Condition="'$(CopyRazorGenerateFilesToPublishDirectory)'==''">false</CopyRazorGenerateFilesToPublishDirectory>
+    <CopyRazorGenerateFilesToPublishDirectory Condition="'$(MvcRazorExcludeViewFilesFromPublish)'=='true'">false</CopyRazorGenerateFilesToPublishDirectory>
+    <CopyRazorGenerateFilesToPublishDirectory Condition="'$(MvcRazorExcludeViewFilesFromPublish)'=='false'">true</CopyRazorGenerateFilesToPublishDirectory>
+
+    <CopyRefAssembliesToPublishDirectory Condition="'$(MvcRazorExcludeRefAssembliesFromPublish)'=='true'">false</CopyRefAssembliesToPublishDirectory>
+    <CopyRefAssembliesToPublishDirectory Condition="'$(MvcRazorExcludeRefAssembliesFromPublish)'=='false'">true</CopyRefAssembliesToPublishDirectory>
```

Comment:
These properties are inverted from the old ones. I based the naming and style of the new ones on other similar settings in the common.targets